docs/jax.config.rst

```diff
@@ -0,0 +1,20 @@
+.. currentmodule:: jax
+
+JAX configuration
+=================
+
+.. autosummary::
+   :toctree: _autosummary
+
+   config
+   enable_checks
+   check_tracer_leaks
+   checking_leaks
```

Comment:
This one is redundant with `check_tracer_leaks`. Maybe just have one or the other? (I'm not sure which though! @LenaMartens WDYT?)

Reply:
This is the full list of functionality imported into the `jax` namespace, so I'd lean toward keeping both here (unless we want to clean up the namespace imports).

Reply:
I think I like `check_tracer_leaks` better because it's more in-line with other debug names (like `debug_nans` and `enable_checks`)? Which one can be used as a context manager again?

Reply:
Both can be used as a context manager; `checking_leaks` is just a partial with the target set to `True`: https://github.com/google/jax/blob/5084a12034e6e88cd1fe274f78a9f58e2d109eeb/jax/_src/config.py#L475